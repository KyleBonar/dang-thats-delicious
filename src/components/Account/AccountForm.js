import React, { Component } from "react";
import PropTypes from "prop-types";
import Validator from "validator";
import TextInput from "../TextInput/TextInput";

class AccountForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: {
        name: "",
        email: ""
      },
      errors: {
        name: "",
        email: ""
      }
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ data: nextProps });
  }

  render() {
    const { name, email } = this.state.data;
    return (
      <form className="form" onSubmit={this.onSubmit}>
        <h2>Settings</h2>

        <TextInput
          type="text"
          id="name"
          name="Name"
          value={name}
          onChange={this.handleNameChange}
        />
        {this.state.errors.name && (
          <p className="form-error">*{this.state.errors.name}</p>
        )}

        <TextInput
          type="email"
          id="email"
          name="Email"
          value={email}
          onChange={this.handleEmailChange}
        />
        {this.state.errors.email && (
          <p className="form-error">*{this.state.errors.email}</p>
        )}

        <button type="submit" className="button button--submit">
          Save ->
        </button>
      </form>
    );
  }

  onSubmit = e => {
    e.preventDefault();

    const errors = this.validate(this.state.data);
    this.setState({ errors });

    if (errors.name.length === 0 && errors.email.length === 0) {
      this.props.onSubmit(this.state.data);
    }
  };

  validate = data => {
    const errors = {};
    errors.name = !data.name ? "Cannot be empty" : "";
    errors.email = !Validator.isEmail(data.email) ? "Invalid email" : "";
    return errors;
  };

  handleNameChange = e => {
    this.setState({ data: { ...this.state.data, name: e.target.value } });
  };

  handleEmailChange = e => {
    this.setState({ data: { ...this.state.data, email: e.target.value } });
  };
}

export default AccountForm;
