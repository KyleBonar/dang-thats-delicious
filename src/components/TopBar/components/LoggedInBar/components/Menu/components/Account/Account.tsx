import * as React from "react";

import Title from "../Title/Title";
import Item from "../Item/Item";

export interface AccountProps {}

const Account: React.FC<AccountProps> = props => {
  const _title = "Account";
  return (
    <>
      <Title>{_title}</Title>
      <Item to="/account/settings">Settings</Item>
      <Item to="#">Profile (Coming Soon)</Item>
    </>
  );
};

export default Account;
