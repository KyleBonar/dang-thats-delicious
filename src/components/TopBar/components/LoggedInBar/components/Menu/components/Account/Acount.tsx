import * as React from "react";

import Title from "../Title/Title";
import Item from "../Item/Item";

export interface AccountProps {}

class Account extends React.PureComponent<AccountProps, any> {
  public render() {
    return (
      <div>
        <Title>Account</Title>
        <Item to="/account/settings">Settings</Item>
        <Item to="#">Profile (Coming Soon)</Item>
      </div>
    );
  }
}

export default Account;
