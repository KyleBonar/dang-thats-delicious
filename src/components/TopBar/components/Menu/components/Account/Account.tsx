import * as React from "react";
import styled from "styled-components";
import Title from "../Title/Title";

export interface AccountProps {}

class Account extends React.PureComponent<AccountProps, any> {
  public render() {
    return (
      <div>
        <Title>Account</Title>
      </div>
    );
  }
}

export default Account;
