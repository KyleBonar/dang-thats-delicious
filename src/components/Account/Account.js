import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import AccountForm from "./AccountForm";
import api from "../../api/api";
import { flashError, flashClose } from "../../actions/flash";
import { getInfo, updateUser } from "../../actions/user";

class Account extends Component {
  componentDidMount() {
    this.getUserInfo();
  }

  render() {
    //check if state exists at all first
    const { name = "", email = "" } = this.props.user;
    return (
      <div className="inner">
        <AccountForm name={name} email={email} onSubmit={this.onSubmit} />
      </div>
    );
  }

  getUserInfo = () => {
    const data = { token: this.props.user.token };

    this.props
      .getInfo(data)
      .then(res => {
        //do something here...?
      })
      .catch(err => {
        this.props.flashError({ text: err.response.data.msg });
      });
  };

  onSubmit = data => {
    //close flash
    this.props.flashClose();

    //add token to data so we can look up user
    data = { ...data, token: this.props.user.token };
    this.props.updateUser(data).catch(err => {
      this.props.flashError({ text: err.response.data.msg });
    });
  };
}

Account.propType = {
  getInfo: PropTypes.func.isRequired,
  updateUser: PropTypes.func.isRequired,
  flashError: PropTypes.func.isRequired,
  flashClose: PropTypes.func.isRequired,
  history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired
};

function mapStateToProps(state) {
  return {
    user: {
      email: state.user.email,
      name: state.user.name,
      token: state.user.token
    }
  };
}

const mapDispatchToProps = {
  getInfo,
  updateUser,
  flashError,
  flashClose
};

export default connect(mapStateToProps, mapDispatchToProps)(Account);
