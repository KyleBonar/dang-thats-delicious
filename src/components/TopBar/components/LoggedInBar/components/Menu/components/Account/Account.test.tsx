import * as React from "react";
import * as enzyme from "enzyme";

import Account from "./Account";

describe("<Account />", () => {
  it("renders", () => {
    const wrapper = enzyme.shallow(<Account />);

    expect(wrapper).toBeTruthy();
  });

  it("matches snapshot", () => {
    const wrapper = enzyme.shallow(<Account />);

    expect(wrapper).toMatchSnapshot();
  });
});
