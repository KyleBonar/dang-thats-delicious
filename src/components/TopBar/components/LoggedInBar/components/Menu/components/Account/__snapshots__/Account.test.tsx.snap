// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account /> matches snapshot 1`] = `
<Fragment>
  <Title>
    Account
  </Title>
  <Item
    to="/account/settings"
  >
    Settings
  </Item>
  <Item
    to="#"
  >
    Profile (Coming Soon)
  </Item>
</Fragment>
`;
